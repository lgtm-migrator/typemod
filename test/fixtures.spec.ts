import fs from 'fs'
import path from 'path'

import type { Transform } from 'jscodeshift'
import jscodeshift from 'jscodeshift'

describe('fixtures', () => {
  const fixturesPath = path.resolve(__dirname, 'fixtures')

  for (const fixturePath of fs.readdirSync(fixturesPath)) {
    if (fixturePath.startsWith('_')) {
      continue
    }

    const fixture = fixturePath.replace(/\.ts$/, '')
    it(`${fixture} should just work`, async () => {
      const { default: transform, parser } = (await import(
        `typemod/${fixture}`
      )) as {
        default: Transform
        parser: 'ts' | 'tsx'
      }

      if (!['ts', 'tsx'].includes(parser)) {
        throw new Error(`${fixture} must have a ts or tsx parser`)
      }

      const j = jscodeshift.withParser(parser)

      const filePath = path.resolve(fixturesPath, fixturePath)

      expect(
        transform(
          {
            path: filePath,
            source: await fs.promises.readFile(filePath, 'utf8'),
          },
          {
            j,
            jscodeshift: j,
            stats: console.log,
            report: console.log,
          },
          {},
        ),
      ).toMatchSnapshot()
    })
  }
})
