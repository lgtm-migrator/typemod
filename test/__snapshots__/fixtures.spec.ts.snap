// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures override should just work 1`] = `
"import { A } from './_override'

export class B extends A {
  // @ts-expect-error
  override keyX = 1

  override keyY = 'y'

  // @ts-expect-error
  override methodX() {
    return 'b'
  }

  // @ts-expect-error
  override methodY = () => 'b'
}
"
`;
